hollow-explorer-ui/src/main/java/com/netflix/hollow/explorer/ui/pages/BrowseSelectedTypePage.java

```diff
@@ -94,14 +94,15 @@ protected void setUpContext(HttpServletRequest req, HollowUISession session, Vel
         String key = req.getParameter("key") == null ? "" : req.getParameter("key");
         Object parsedKey[] = null;
         try {
-            parsedKey = parseKey(primaryKey, key);
+            parsedKey = parseKey(ui.getStateEngine(), primaryKey, key);
         } catch(Exception e) {
             key = "";
         }
 
-        Integer ordinal =
-            getOrdinalToDisplay(req, key, parsedKey, selectedOrdinals, fieldPathIndexes);
-        if (ordinal != null && !ordinal.equals(-1) && "".equals(key)
+        HollowTypeReadState readTypeState = getTypeState(req);
+        int ordinal = req.getParameter("ordinal") == null ? -1 : Integer.parseInt(req.getParameter("ordinal"));
```

Comment:
Minor: it would be great if there is a constant for all those  -1.

Reply:
done